### ISSUE TITLE
useAsyncData - blocked page

### ISSUE THREAD

**Issue by aslero** [nuxt/framework#6199]
### Environment

useAsyncData

### Reproduction

useAsyncData

### Describe the bug

Hello. I created a repository

```
async getExcursionsByPaginate(params): Promise<any> {
    return await useAsyncData('excursions', () =>
      $fetch(this.urlGenerator.getExcursionsByPaginate(), {
        baseURL: this.config.API_BASE_URL,
        params
      })
    );
  }
```

on the page I call like this

```
<template>
  <div>
          <SkeletonExcursionIndex v-if="pending" />
          <template v-else>
            <div
              v-for="excursion in excursions"
              :key="excursion.id"
              class="tw-pb-8"
            >
              <ExcursionsFilterItem
                :key="excursion.id"
                :excursion="excursion"
              />
            </div>
          </template>
  </div>
</template>

<script lang="ts" setup>
import { inject } from '#imports';

const repositoryService = inject('repositoryService');

const excursions = ref([]);

const {
  pending,
  data: result
} =
  await repositoryService.repository.getExcursionsByPaginate({ take: 50 });

excursions.value = result.value.data;

useHead({
  title: 'test',
  meta: [
    {
      name: 'description',
      content: 'test'
    }
  ]
});
</script>

```

but when you go to the page, there is a delay of about 2 seconds and at this time the previous page is shown, after this delay the page to which the data was also shown is shown, but pending does not work and I do not see the skeleton component

**Comment by danielroe** [comment]
If you want to show a skeleton component, then you can either use `useLazyAsyncData` or pass `{ lazy: true }` as an option to `useAsyncData`.

**Comment by aslero** [comment]
But how to get the data and then work with them?

**Comment by danielroe** [comment]
You can watch the data within script setup, and of course the template will be reactive.

**Comment by aslero** [comment]
How is that?

**Comment by aslero** [comment]
Where I reload the page no data is output

```
const excursions = ref([]);

const {
  pending,
  data: query
} =
   repositoryService.repository.getExcursionsByPaginate(params);

watch(query, () => {
  excursions.value = query.value.data;
});
```

**Comment by aslero** [comment]
It worked like this, but in my opinion this is a crooked and wrong decision?

```
const excursions = ref([]);

const {
  pending,
  data: result
} =
  await repositoryService.repository.getExcursionsByPaginate(params);

excursions.value = result.value.data

watch(result, () => {
  excursions.value = result.value.data;
});
```

**Comment by aslero** [comment]
and writes an error

```
index.vue:67 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'data')
    at index.vue:67:1
    at withAsyncContext (runtime-core.esm-bundler.js:7560:21)
    at setup (index.vue:67:1)
```

**Comment by aslero** [comment]
if initialCache: false, then when you go to the page, it shows data, but when reloading, the data is not displayed, if initialCache: true, then no data is displayed at all

```
const {
  pending,
  data: result
} =
  repositoryService.repository.getExcursionsByPaginate(params);

watch(result, (newResult) => {
  excursions.value = result?.value?.data;
  meta.total = result?.value?.meta?.total ?? 0;
});
```

**Comment by danielroe** [comment]
If you provide a minimal sandbox, I'll show you how.

**Comment by aslero** [comment]
send invite

> If you provide a minimal sandbox, I'll show you how.

**Comment by danielroe** [comment]
There you go :+1:

tldr; move imperative statements to computed properties.

**Comment by aslero** [comment]
> There you go (codesandbox edited) 👍
> 
> tldr; move imperative statements to computed properties.

Good) but paginate not work https://github.com/nuxt/framework/issues/6273#issue-1323938016

**Comment by aslero** [comment]
> There you go (codesandbox edited) 👍
> 
> tldr; move imperative statements to computed properties.

and 
![Uploading Снимок экрана 2022-08-02 в 13.30.16.png…]()

**Comment by danielroe** [comment]
Seems to be working fine for me in the sandbox. Would you clarify what isn't working? https://codesandbox.io/s/sleepy-butterfly-x9vsh5.

**Comment by aslero** [comment]
> Кажется, у меня все работает в песочнице. Не могли бы вы уточнить, что не работает? https://codesandbox.io/s/sleepy-butterfly-x9vsh5 .

This is strange, everything works in the sandbox, but pagination does not work on my local

**Comment by aslero** [comment]
> Seems to be working fine for me in the sandbox. Would you clarify what isn't working? https://codesandbox.io/s/sleepy-butterfly-x9vsh5.

Sent an invite to the repository
